checks/cs50/2017/x/finance/checks.py

```diff
@@ -1,13 +1,305 @@
+                field.send_keys(password)
+        if fields_filled != 2:
+            raise Error("Could not fill in username and/or password.")
+
+        # search for log in button
+        buttons = self.driver.find_elements_by_tag_name("button")
+        login_button = None
+        for button in buttons:
+            if button.text == "Log In":
+                login_button = button
+        if login_button == None:
+            raise Error("Could not find log in button.")
+        login_button.click()
+
+        if "Apology" in self.driver.title:
+            raise Error("Expected login to succeed, not fail.")
+
+    def quote(self, ticker, outcome, value=None):
+        """Checks that getting a quote results in desired outcome."""
+
+        # go to quote page
+        self.driver.get("{}/quote".format(Finance.BASE_PATH))
+        fields = self.driver.find_elements_by_tag_name("input")
     
+        # find field to input quote
+        quote_field = None
+        for field in fields:
+            name = field.get_attribute("name")
+            if name == "symbol":
+                quote_field = field
+        if quote_field == None:
+            raise Error("Could not input quote.")
+        quote_field.send_keys(ticker)
+
+        # search for quote button
+        buttons = self.driver.find_elements_by_tag_name("button")
+        quote_button = None
+        for button in buttons:
+            if button.text == "Quote":
+                quote_button = button
+        if quote_button == None:
+            raise Error("Could not find quote button.")
+        quote_button.click()
+
+        if "Apology" in self.driver.title and outcome:
+            raise Error("Expected quote success, not failure.")
+        elif "Apology" not in self.driver.title and not outcome:
+            raise Error("Expected quote failure, not success.")
+        elif outcome and value:
+            # check the output against the given value
+            if str(value) not in self.driver.find_element_by_tag_name("body").text:
+                raise Error("Expected quote not found.")
+
+    def transaction(self, transaction, symbol, shares, outcome, value=None, remaining=None):
+        """Checks that purchasing shares results in given outcome."""
+
+        # go to buy page
+        self.driver.get("{}/{}".format(Finance.BASE_PATH, transaction))
+        fields = self.driver.find_elements_by_tag_name("input")
+
+        # fill in fields
+        fields_filled = 0
+        for field in fields:
+            name = field.get_attribute("name")
+            if name == "symbol":
+                fields_filled += 1
+                field.send_keys(symbol)
+            elif name == "shares":
+                fields_filled += 1
+                field.send_keys(shares)
+        if fields_filled != 2:
+            raise Error("Could not fill in symbol and/or shares.")
+
+        # search for buy button
+        buttons = self.driver.find_elements_by_tag_name("button")
```

Comment:
Should we wrap these most commonly used `driver` methods (this one, get, post, etc.) so that we can add automatic logging to explain to students what's going on?